### ISSUE TITLE
Imenu in python buffers stops working without semantic layer

### ISSUE THREAD

**Issue by bmag** [syl20bnr/spacemacs#2569]
Steps to reproduce:
1) make sure python layer is enabled and semantic layer is disabled in your `.spacemacs` file
2) start spacemacs
3) open a python file
4) hit <kbd>SPC s l</kbd> (`spacemacs/jump-in-buffer`). A helm buffer with some imenu entries pops up.
5) close the helm buffer and open a new python file
6) hit <kbd>SPC s l</kbd>. Expected behavior: same as in step 4. Actual behavior: there are no imenu entries.

Looks like imenu can't find any entries in any python file I open from here on. I've checked the value of `imenu-create-index-function`, which generates the imenu entries. In the first file its value is `python-imenu-create-index`, but in the second file its value is `semantic-create-imenu-index` - despite having disabled the semantic layer.

If I have the semantic layer enabled, I don't encounter this bug.

I'll research this bug further later.

**Comment by bmag** [comment]
Turns out this is an emacs bug. To reproduce:
1) emacs -Q
2) `M-: (require 'semantic)`
3) Open a python file: no entries for imenu, `imenu-create-index-function` is `semantic-create-imenu-index`.

For some reason, loading `semantic` also loads `semantic/wisent/python`, which includes these two lines:
```elisp
;;;###autoload
(add-hook 'python-mode-hook 'wisent-python-default-setup)
```
But `wisent-python-default-setup` doesn't check if `semantic-mode` is on or off.
I'll see if I can report it upstream or provide a work-around.

**Comment by tuhdo** [comment]
It's not a bug. Wisen is part of Semantic and is only loaded when a user loads Semantic.

**Comment by bmag** [comment]
Spacemacs' standard interface to imenu <kbd>SPC s l</kbd>. That interface loads semantic. Loading semantic prevents imenu from working in new python buffers. With all due respect, how is that not a bug?

**Comment by tuhdo** [comment]
I mean it's not an Emacs bug. It's simply how i works. Spacemacs uses a command not in Emacs, so it's the comand that is bugged.

**Comment by fgallina** [comment]
Hi, just a heads up.

I just pushed a fix[0] in Emacs trunk for this after this bug report[1].  A workaround for the time being would be something along these lines:

    (eval-after-load "semantic"
      (remove-hook 'python-mode-hook 'wisent-python-default-setup))

[0] http://git.savannah.gnu.org/cgit/emacs.git/commit/?id=bf0ed7340098dcb6228566450a99d3cbc0460094
[1] https://debbugs.gnu.org/cgi/bugreport.cgi?bug=21220

**Comment by bmag** [comment]
@fgallina thank you

**Comment by bmag** [comment]
The fix introduced by #2583 was removed at some point, and this issue is happening again, so I'm reopening it :frowning_face:

**Comment by TheBB** [comment]
Should be fixed in master.
